packages/uicore/src/components/ExpressionDropdown/ExpressionDropdown.tsx

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright 2023 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+import React, { useState } from 'react'
+import { Boundary, Card, Menu, OverflowList, Popover } from '@blueprintjs/core'
+import { Icon, IconName } from '@harness/icons'
+import { Layout } from '../../layouts/Layout'
+import { isEmpty, isUndefined } from 'lodash-es'
+import css from './ExpressionDropdown.css'
+
+export interface ChildKeyNode {
+  key: string
+  value: string
+}
+
+interface TrieNode {
+  value: string
+  valueTillHere: string
+  children: TrieNode[]
+  childKeys: ChildKeyNode[]
+  childExpressions: string[]
+}
+
+interface NewExpressionDropdownProps {
+  rootTrieNode: TrieNode
+  setQueryValue: React.Dispatch<React.SetStateAction<string>>
+  setInputValue: React.Dispatch<React.SetStateAction<string>>
+  query: string
+  itemRenderer: React.ReactNode
+}
+
+interface getVisibleItemRendererProps {
+  dropDownItemClickHandler: (value: string) => void
+  isOpen: boolean[]
+  setIsOpen: React.Dispatch<React.SetStateAction<boolean[]>>
+}
+
+function getTargetElement(
```

Comment:
1. move to utils
2. Instead of passing setter here, return value and set in component

Reply:
3. func name does not match what it is doing

Reply:
Moved to utils , Renamed as `setOpenTargetElement` , still setting the value inside the util as we need this at multiple places